cypress-io/cypress
# fix: add lib/cli to package.json exports

### User facing changelog

n/a

### Additional details

Bazel `rules_nodejs` repository is providing [a Bazel example for Cypress 7+](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/cypress). It demonstrates how `@bazel/cypress` can be used to run Cypress tests using the Bazel build system. However it relies on a file, `run-cypress.js`, which imports [`cypress/lib/cli`](https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/cypress/internal/run-cypress.js#L1). 

With Cypress 10+, this causes an error `ERR_PACKAGE_PATH_NOT_EXPORTED`.

> Package subpath './lib/cli' is not defined by "exports" in /private/var/tmp/_bazel_[.....]/yarn/node_modules/cypress/package.json

**This is not surprising, as [Cypress 7 was not using `exports` entries](https://github.com/cypress-io/cypress/blob/v7.7.0/cli/package.json).**

Adding the proper entry to `exports` in `cli/package.json` solves the problem. And the Cypress tests are properly executed.

### Steps to test


- Clone the `https://github.com/bazelbuild/rules_nodejs`
- `cd` to examples/cypress
- `npm i -g @bazel/bazelisk` (in case you don't have bazel)
- `yarn test`

See it's working.

- Now use `yarn` to upgrade to Cypress 10+.
- Repeat the operations
- See that it fails with `ERR_PACKAGE_PATH_NOT_EXPORTED`

Getting into the execroot of Bazel and changing the package.json in `node_modules/cypress` to add this new entry to `exports` fix the error.

### How has the user experience changed?

n/a

### PR Tasks
